data-access/Models/UserModels.cs

```diff
@@ -51,7 +51,14 @@ public UserInfo(string requestPath, IUserRepository userRepository)
                 var scrollEditionMatch = scrollEditionTest.Match(_requestPath);
                 if (scrollEditionMatch.Groups.Count == 2)
                     _editionId = uint.TryParse(scrollEditionMatch.Groups[1].Value, out var i) ? i : 0;
-                else _editionId = 0;
+                else
+                {
+                    scrollEditionTest = new Regex(@"^.*/editions/(\d{1,32}).*$");
+                    scrollEditionMatch = scrollEditionTest.Match(_requestPath);
+                    if (scrollEditionMatch.Groups.Count == 2)
+                        _editionId = uint.TryParse(scrollEditionMatch.Groups[1].Value, out var i) ? i : 0;
+                    else _editionId = 0;
+                }
```

Comment:
Is _requestPath the URL passed to the controller? The model shouldn't parse it in any way. If we change the URL (drop the /editions, for example, or rename /editions to /edition), the data model shouldn't care. If you need the edition ID, add is a a parameter, and let the caller parse the URL.

Reply:
Yes, this already caused me trouble once.  I am fixing it now as you suggest.

Reply:
Ok. It is finished now. Thanks.